src/Processors/QueryPlan/JoinStepLogical.h

```diff
@@ -90,16 +92,19 @@ class JoinStepLogical final : public IQueryPlanStep
     JoinExpressionActions expression_actions;
     JoinInfo join_info;
 
-    bool swap_inputs = false;
     Names required_output_columns;
 
-    PreparedJoinStorage prepared_join_storage;
-    IQueryTreeNode::HashState hash_table_key_hash;
+    bool use_nulls;
```

Comment:
This flag looks very important to me cause it might change the output type.
So, I decided to move it from settings to here.